File: Backend_Springboot/src/main/java/com/hostel_mng_sys/service/AppointmentService.java
package com.hostel_mng_sys.service;

import java.util.List;

import com.hostel_mng_sys.login.entity.Appointment;

public interface AppointmentService {

	public Appointment createAppointment (Appointment appointment);
	public List<Appointment> getAllAppointments();
	public void deleteAppointment(Long id);
}

File: Frontend_Angular/src/app/appointment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Appointment } from './appointment';

@Injectable({
  providedIn: 'root'
})
export class AppointmentService {

  constructor(private httpClient : HttpClient) { }

  private baseUrl = "http://localhost:8080/api/allappointments"
  getAllAppointments(): Observable<Appointment[]>{
    return this.httpClient.get<Appointment[]>(`${this.baseUrl}`)

  }
  private addUrl = "http://localhost:8080/api/appointment"
  createAppointment(appointment:Appointment):Observable<Appointment>{

    return this.httpClient.post<Appointment>(`${this.addUrl}`,appointment);
  }
  private delUrl = "http://localhost:8080/api/delappt"
  deleteAppointment(id:number):Observable<object>{

    return this.httpClient.delete(`${this.delUrl}/${id}`);
  }
}
